src/components/Badge/Badge.test.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import * as renderer from 'react-test-renderer';
+
+import { Badge } from '.';
+
+describe('Badge component', () => {
+  it("should render empty badge label with primary background color when we don't pass any props", () => {
+    const tree = renderer.create(<Badge />);
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should render Badge component with text when we pass children', () => {
+    const tree = renderer.create(<Badge>Simple badge</Badge>);
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should render Badge component with text and Secondary background color when we pass children and color props', () => {
+    const tree = renderer.create(<Badge color="secondary">Simple badge</Badge>);
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should render Badge component with capitalized first letter', () => {
+    const tree = renderer.create(<Badge>simple badge</Badge>);
+    expect(tree).toMatchSnapshot();
+  });
+});
```

Comment:
I think adding test scenarios other than snapshot will add more value 

### Possible scenarios 

- The rendered text of the Badge component should have the first letter capitalized 
- In case we used the component with out text, it should not render any text

Reply:
I guess in our case, it's fine to use only snapshot tests as it's a dumb component,
and the cases which you've mentioned are already covered in snapshot tests

Reply:
You are right, snapshot will make it easily to compare changes visually.
but some cases the test will fail even if the business requirement  did change 

> Ex : if you call the component` <Badge> Text </Badge>` instead of ` <Badge>Text</Badge>`  the test will fail as there will be extra space before and after the `Text`.


In the case to make the test robust we need to ignore spaces of the the innerText. I do not know if we can achieve it with snap shoot testing.